shared/p2p/hobbits/rpc_handler.go

```diff
@@ -0,0 +1,296 @@
+package hobbits
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/go-ssz"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	v1alpha1 "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/p2p"
+	"github.com/renaynay/go-hobbits/encoding"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type RPCHeader struct {
+	MethodID uint16 `bson:"method_id"`
+	Id       uint64 `bson:"id"`
+}
+
+type Hello struct {
```

Comment:
Consider breaking out all of these types into their own file `types.go`. It would make for better logic separation.